modules/repository-url/src/main/java/org/elasticsearch/common/blobstore/url/URLBlobStore.java

```diff
@@ -40,10 +42,22 @@
      * @param settings settings
      * @param path     base URL
      */
-    public URLBlobStore(Settings settings, URL path) {
+    public URLBlobStore(Settings settings, URL path, URLHttpClient httpClient) {
         this.path = path;
         this.bufferSizeInBytes = (int) settings.getAsBytesSize("repositories.uri.buffer_size",
             new ByteSizeValue(100, ByteSizeUnit.KB)).getBytes();
+        int maxRetries = settings.getAsInt("repositories.uri.http.max_retries", HttpClientSettings.DEFAULT_MAX_RETRIES);
```

Comment:
Document these new settings?

Also best to define these as proper Setting objects

Reply:
Addressed in https://github.com/elastic/elasticsearch/pull/69521/commits/9e99fac203884a97b3a6386306e1cff870a86960